app/controllers/declaration/RoutingCountriesSummaryController.scala

```diff
@@ -93,14 +94,18 @@ class RoutingCountriesSummaryController @Inject()(
       .fold(
         formWithErrors => Future.successful(BadRequest(removeRoutingCountryPage(mode, formWithErrors, country))),
         validAnswer =>
-          if (validAnswer) {
-            removeCountry(Country(Some(countryCode))).map { _ =>
-              navigator.continueTo(mode, controllers.declaration.routes.RoutingCountriesSummaryController.displayPage)
-            }
-          } else Future.successful(navigator.continueTo(mode, controllers.declaration.routes.RoutingCountriesSummaryController.displayPage))
+          if (validAnswer) removeCountry(Country(Some(countryCode))).map(_ => removeRedirect(mode))
+          else Future.successful(removeRedirect(mode))
       )
   }
 
+  private def removeRedirect(mode: Mode)(implicit request: JourneyRequest[AnyContent]): Result =
+    if (mode == ErrorFix) {
+      Redirect(controllers.declaration.routes.RoutingCountriesSummaryController.displayPage(mode))
+    } else {
+      navigator.continueTo(mode, controllers.declaration.routes.RoutingCountriesSummaryController.displayPage)
```

Comment:
This is the 2nd or 3rd time you've done this.  On the face of it it looks like both paths lead to the same re-direct (and so the code os hard to understand).   Can't this logic be added to Navigator.continueTo?